index df7082b6..0022e5fa 100644
--- a/OpenRGB.pro
+++ b/OpenRGB.pro
@@ -588,9 +588,9 @@ contains(QMAKE_PLATFORM, linux) {
     icon.files+=qt/org.openrgb.OpenRGB.png
     metainfo.path=$$PREFIX/share/metainfo/
     metainfo.files+=qt/org.openrgb.OpenRGB.metainfo.xml
-    systemd_service.path=/etc/systemd/system
-    systemd_service.files+=qt/openrgb.service
-    INSTALLS += target desktop icon metainfo udev_rules systemd_service
+    # systemd_service.path=/etc/systemd/system
+    # systemd_service.files+=qt/openrgb.service
+    INSTALLS += target desktop icon metainfo udev_rules # systemd_service
 }
