diff --git a/README.md b/README.md
index 0868434..c70735c 100644
--- a/README.md
+++ b/README.md
@@ -365,6 +365,11 @@ default, you must create it manually.
             "wallpapers": false
         },
         "showOnHover": false
+        "hiddenApps": [
+            "fish",
+            "qt5ct",
+            "qt6ct",
+        ],
     },
     "lock": {
         "recolourLogo": false
diff --git a/config/LauncherConfig.qml b/config/LauncherConfig.qml
index d021089..1a518d5 100644
--- a/config/LauncherConfig.qml
+++ b/config/LauncherConfig.qml
@@ -10,6 +10,7 @@ JsonObject {
     property bool enableDangerousActions: false // Allow actions that can cause losing data, like shutdown, reboot and logout
     property int dragThreshold: 50
     property bool vimKeybinds: false
+    property list<string> hiddenApps: []
     property UseFuzzy useFuzzy: UseFuzzy {}
     property Sizes sizes: Sizes {}
 
diff --git a/modules/launcher/services/Apps.qml b/modules/launcher/services/Apps.qml
index 9f87d50..5ee2b0c 100644
--- a/modules/launcher/services/Apps.qml
+++ b/modules/launcher/services/Apps.qml
@@ -69,7 +69,9 @@ Searcher {
     Variants {
         id: variants
 
-        model: [...DesktopEntries.applications.values].sort((a, b) => a.name.localeCompare(b.name))
+        model: [...DesktopEntries.applications.values]
+            .filter(a => !Config.launcher.hiddenApps.includes(a.id))
+            .sort((a, b) => a.name.localeCompare(b.name))
 
         QtObject {
             required property DesktopEntry modelData
