diff --git a/modules/bar/Bar.qml b/modules/bar/Bar.qml
index 22f21dd..f705cbf 100644
--- a/modules/bar/Bar.qml
+++ b/modules/bar/Bar.qml
@@ -113,6 +113,7 @@ Item {
                 id: workspacesInner
 
                 anchors.centerIn: parent
+                monitorName: screen.name
             }
         }
 
diff --git a/modules/bar/components/workspaces/ActiveIndicator.qml b/modules/bar/components/workspaces/ActiveIndicator.qml
index b44ba87..b18cac2 100644
--- a/modules/bar/components/workspaces/ActiveIndicator.qml
+++ b/modules/bar/components/workspaces/ActiveIndicator.qml
@@ -7,13 +7,14 @@ import QtQuick
 StyledRect {
     id: root
 
+    required property int activeWsId
     required property list<Workspace> workspaces
     required property Item mask
     required property real maskWidth
     required property real maskHeight
     required property int groupOffset
 
-    readonly property int currentWsIdx: Hyprland.activeWsId - 1 - groupOffset
+    readonly property int currentWsIdx: activeWsId - 1 - groupOffset
     property real leading: getWsY(currentWsIdx)
     property real trailing: getWsY(currentWsIdx)
     property real currentSize: workspaces[currentWsIdx]?.size ?? 0
diff --git a/modules/bar/components/workspaces/Workspace.qml b/modules/bar/components/workspaces/Workspace.qml
index f5b99f5..67fb1e7 100644
--- a/modules/bar/components/workspaces/Workspace.qml
+++ b/modules/bar/components/workspaces/Workspace.qml
@@ -10,6 +10,7 @@ Item {
     id: root
 
     required property int index
+    required property int activeWsId
     required property var occupied
     required property int groupOffset
 
@@ -32,8 +33,8 @@ Item {
         readonly property string activeLabel: Config.bar.workspaces.activeLabel || (root.isOccupied ? occupiedLabel : label)
 
         animate: true
-        text: Hyprland.activeWsId === root.ws ? activeLabel : root.isOccupied ? occupiedLabel : label
-        color: Config.bar.workspaces.occupiedBg || root.isOccupied || Hyprland.activeWsId === root.ws ? Colours.palette.m3onSurface : Colours.layer(Colours.palette.m3outlineVariant, 2)
+        text: activeWsId === root.ws ? activeLabel : root.isOccupied ? occupiedLabel : label
+        color: Config.bar.workspaces.occupiedBg || root.isOccupied || activeWsId === root.ws ? Colours.palette.m3onSurface : Colours.layer(Colours.palette.m3outlineVariant, 2)
         horizontalAlignment: StyledText.AlignHCenter
         verticalAlignment: StyledText.AlignVCenter
 
diff --git a/modules/bar/components/workspaces/Workspaces.qml b/modules/bar/components/workspaces/Workspaces.qml
index 0a806f7..34a36b6 100644
--- a/modules/bar/components/workspaces/Workspaces.qml
+++ b/modules/bar/components/workspaces/Workspaces.qml
@@ -8,12 +8,15 @@ import QtQuick.Layouts
 Item {
     id: root
 
+    required property string monitorName
+
+    readonly property int activeWsId: Hyprland.getActiveWorkspaceForMonitor(monitorName)
     readonly property list<Workspace> workspaces: layout.children.filter(c => c.isWorkspace).sort((w1, w2) => w1.ws - w2.ws)
     readonly property var occupied: Hyprland.workspaces.values.reduce((acc, curr) => {
         acc[curr.id] = curr.lastIpcObject.windows > 0;
         return acc;
     }, {})
-    readonly property int groupOffset: Math.floor((Hyprland.activeWsId - 1) / Config.bar.workspaces.shown) * Config.bar.workspaces.shown
+    readonly property int groupOffset: Math.floor((activeWsId - 1) / Config.bar.workspaces.shown) * Config.bar.workspaces.shown
 
     implicitWidth: layout.implicitWidth
     implicitHeight: layout.implicitHeight
@@ -29,6 +32,7 @@ Item {
             model: Config.bar.workspaces.shown
 
             Workspace {
+                activeWsId: root.activeWsId
                 occupied: root.occupied
                 groupOffset: root.groupOffset
             }
@@ -54,6 +58,7 @@ Item {
         asynchronous: true
 
         sourceComponent: ActiveIndicator {
+            activeWsId: root.activeWsId
             workspaces: root.workspaces
             mask: layout
             maskWidth: root.width
diff --git a/services/Hyprland.qml b/services/Hyprland.qml
index 7bb9458..930be4e 100644
--- a/services/Hyprland.qml
+++ b/services/Hyprland.qml
@@ -21,6 +21,11 @@ Singleton {
         Hyprland.dispatch(request);
     }
 
+    function getActiveWorkspaceForMonitor(monitorName) {
+        const monitor = monitors.values.find(m => m.name === monitorName);
+        return monitor ? monitor.lastIpcObject.activeWorkspace.id : -1;
+    }
+
     Connections {
         target: Hyprland
 
