diff --git a/modules/drawers/Interactions.qml b/modules/drawers/Interactions.qml
index fcaa790..a238172 100644
--- a/modules/drawers/Interactions.qml
+++ b/modules/drawers/Interactions.qml
@@ -71,8 +71,7 @@ CustomMouseArea {
             if (!utilitiesShortcutActive)
                 visibilities.utilities = false;
 
-            if (!popouts.currentName.startsWith("traymenu"))
-                popouts.hasCurrent = false;
+            popouts.hasCurrent = false;
 
             if (Config.bar.showOnHover)
                 bar.isHovered = false;
@@ -166,7 +165,7 @@ CustomMouseArea {
         // Show popouts on hover
         if (x < bar.implicitWidth)
             bar.checkPopout(y);
-        else if (!popouts.currentName.startsWith("traymenu") && !inLeftPanel(panels.popouts, x, y))
+        else if (!inLeftPanel(panels.popouts, x, y))
             popouts.hasCurrent = false;
     }
 
