diff --git a/src/caelestia/utils/wallpaper.py b/src/caelestia/utils/wallpaper.py
index 6c6532c..d5613ab 100644
--- a/src/caelestia/utils/wallpaper.py
+++ b/src/caelestia/utils/wallpaper.py
@@ -125,6 +125,7 @@ def get_colours_for_wall(wall: Path | str, no_smart: bool) -> None:
 
 
 def set_wallpaper(wall: Path | str, no_smart: bool) -> None:
+    print(f"Setting wallpaper to {wall}")
     # Make path absolute
     wall = Path(wall).resolve()
 
@@ -146,17 +147,7 @@ def set_wallpaper(wall: Path | str, no_smart: bool) -> None:
     wallpaper_thumbnail_path.unlink(missing_ok=True)
     wallpaper_thumbnail_path.symlink_to(thumb)
 
-    scheme = get_scheme()
-
-    # Change mode and variant based on wallpaper colour
-    if scheme.name == "dynamic" and not no_smart:
-        smart_opts = get_smart_opts(wall, cache)
-        scheme.mode = smart_opts["mode"]
-        scheme.variant = smart_opts["variant"]
-
-    # Update colours
-    scheme.update_colours()
-    apply_colours(scheme.colours, scheme.mode)
+    print("Done setting wallpaper")
 
 
 def set_random(args: Namespace) -> None:
